### PR TITLE
added support to set render texture as video source

### PR DESCRIPTION
**Pull request by soorin1990**

### CODE REVIEW

**Comment by Yousif-CS** [comment]
Can you fix the conflicts in this branch?

File: Runtime/McPublisher.cs [hunk 1]
```diff
@@ -441,6 +446,79 @@ public void SetVideoSource(Camera source, StreamSize resolution = null)
             }
         }
 
+        /// <summary>
+        /// Set the video source for capture. Call with null as the source to remove
+        /// the currently set video source.
+        /// </summary>
+        /// <param name="source">The Target Render Texture source to capture from</param>
+        /// <param name="resolution">The capturing resolution</param>
+        public void SetVideoSource(RenderTexture source, StreamSize resolution = null)
```
Comment (Yousif-CS) [inline Runtime/McPublisher.cs:436]: Can we refactor the internals of the two overloads of SetVideoSource as they are quite similar. I'd suggest splitting the internals into separate functions to reduce repetition

Comment (soorin1990) [reply Runtime/McPublisher.cs:436]: refactored code to avoid duplicate code.

**Review by Yousif-CS — APPROVED** [review]
LGTM

**Pull request merged**